tests/test_authentication.py

```diff
@@ -207,3 +207,62 @@ def test_login_required_roles_debuggable_but_still_requires_roles(
             mock_request.location
             == "https://authenticator/service/user?roles_required=COMMENTER"
         )
+
+    def test_login_required_with_return_app_redirects_to_signed_out_without_token(
+        self, flask_test_client
+    ):
+        """
+        GIVEN a flask_test_client and
+            route decorated with @login_required decorator with the "return_app" parameter set to "post-award-frontend"
+        WHEN a request is made without any "fsd-user-token" cookie
+        THEN the route redirects to the authenticator /sessions/sign-out url with correct return_app query param
+        :param flask_test_client:
+        """
+        mock_request = flask_test_client.get("/mock_login_requested_return_app_route")
+
+        assert mock_request.status_code == 302
+        assert (
+            mock_request.location
+            == "https://authenticator/sessions/sign-out?return_app=post-award-frontend"
+        )
+
+    def test_login_required_with_return_app_redirects_to_signed_out_with_invalid_token(
+        self, flask_test_client
+    ):
+        """
+        GIVEN a flask_test_client and
+            route decorated with @login_required decorator with the "return_app" parameter set to "post-award-frontend"
+        WHEN a request is made with a correctly formatted
+            but invalidly signed "fsd-user-token" cookie
+        THEN the route redirects to the authenticator /sessions/sign-out url with correct return_app query param
+        :param flask_test_client:
+        """
+        invalid_token = self._create_invalid_token()
+        flask_test_client.set_cookie("localhost", "fsd-user-token", invalid_token)
+        mock_request = flask_test_client.get("/mock_login_requested_return_app_route")
+
+        assert mock_request.status_code == 302
+        assert (
+            mock_request.location
+            == "https://authenticator/sessions/sign-out?return_app=post-award-frontend"
+        )
+
+    def test_login_required_with_return_app__sets_user_attributes_with_valid_token(
```

Comment:
an accidental double underscore here between app and sets?

Reply:
ah nice spot